src/view/states/task_input/SetLabelState.cpp

```diff
@@ -5,22 +5,28 @@
 #include "SetLabelState.h"
 #include "user_interface/console_io/ConsoleUtilities.h"
 
-SetLabelState::SetLabelState(const std::shared_ptr<StatesFactory>& factory) :
-                            factory_(factory) {
+SetLabelState::SetLabelState(const StateType next_state,
+                             const StateType error_state,
+                             const std::shared_ptr<ConsolePrinter>& printer,
+                             const std::shared_ptr<ConsoleReader>& reader,
+                             const std::shared_ptr<CommandFactory>& command_factory) :
+                             next_state_(next_state),
+                             error_state_(error_state),
+                             printer_(printer),
+                             reader_(reader),
+                             command_factory_(command_factory) {}
 
-}
-
-std::shared_ptr<State> SetLabelState::Execute(StateContext& context) {
+StateType SetLabelState::Execute(StateContext& context) {
     std::optional<TaskId> task_id =
-            console_io::util::GetTaskIdFromUser("Task ID", *factory_.lock()->GetConsolePrinter(), *factory_.lock()->GetConsoleReader());
+            console_io::util::GetTaskIdFromUser("Task ID", *printer_, *reader_);
     if (!task_id.has_value()){
-        factory_.lock()->GetConsolePrinter()->WriteError("Incorrect task id was given, try again!");
-        return factory_.lock()->GetNextState(*this, StatesFactory::MoveType::ERROR);
+        printer_->WriteError("Incorrect task id was given, try again!");
+        return error_state_;
     }
 
-    std::string label_to_set = console_io::util::GetUserInput("Label", *factory_.lock()->GetConsolePrinter(), *factory_.lock()->GetConsoleReader());
+    std::string label_to_set = console_io::util::GetUserInput("Label", *printer_, *reader_);
     context.SetTaskId(task_id.value());
```

Comment:
You have already checked if `task_id` contains a value in `22`. Just dereference here.